codeface/R/cluster/graph_comparison.r

```diff
@@ -142,37 +151,248 @@ graph.comparison <- function(g.1, g.2) {
+                         compare.edge.count=numeric(),
+                         compare.vertex.count=numeric(),
+                         compare.cohesion=numeric(),
+                         compare.diameter=numeric(),
+                         compare.density=numeric(),
+                         compare.transitivity=numeric(),
+                         nodes.diff=numeric(),
+                         vertex.weighted.diff=numeric(),
+                         vertex.total.diff=numeric(),
+                         unmerged.vertex.weighted.diff=numeric(),
+                         unmerged.vertex.total.diff=numeric(),
+                         stringsAsFactors=FALSE)
+  get.project.data <- function(con, range) {
+    project.id <- get.project.id.from.release.range.id(con, range)
+    if (is.null(project.id)) stop(str_c("Range id ", range, " is unknown!"))
+    cycle <- get.cycle.from.release.range.id(con, range)
+    project <- get.project.from.project.id(con, project.id)
+    return (list(project.name=project$name, type=project$analysisMethod, range.string=cycle))
+  }
+  # Calculate some graph metrices
+  get.graph.metrices <- function(g) {
+    return (list(
+      edge.count = length(E(g)),
+      vertex.count = length(V(g)),
+      cohesion = graph.cohesion(g),
+      diameter = diameter(g),
+      density = graph.density(g),
+      transitivity = transitivity(g)))
+  }
+
+  for (i in 1:len) {
+    r.1 <- compare.ranges$original[i]
+    r.2 <- compare.ranges$compare[i]
+    meta.data.1 <- get.project.data(con, r.1)
+    meta.data.2 <- get.project.data(con, r.2)
+    print (str_c("Calculating comparison of ", meta.data.1$project.name, "/", meta.data.1$type, "(", meta.data.1$range.string ,") with ",
+                 meta.data.2$project.name, "/", meta.data.2$type, "(", meta.data.2$range.string ,")..."))
+    data.1 <- get.graph.data.from.range(con, r.1)
+    data.2 <- get.graph.data.from.range(con, r.2)
+    igraphs <- get.merged.igraphs(data.1,data.2)
+    diff.sym <- run.graph.comparison(igraphs, symmetric=F)
+    overview[nrow(overview)+1,] <- c(i,
+                                     meta.data.1, meta.data.2,
+                                     get.graph.metrices(igraphs$unmerged.g.1), get.graph.metrices(igraphs$unmerged.g.2),
+                                     diff.sym$nodes.diff, diff.sym$vertex.weighted.diff, diff.sym$vertex.total.diff,
+                                     diff.sym$unmerged.vertex.weighted.diff, diff.sym$unmerged.vertex.total.diff)
+
+    #vertexdata[[as.character(str_c(meta.data.2$type, "_", meta.data.2$range.string))]] <- diff.sym$vertex.diff
+    vertexdata[[as.character(str_c(as.character(r.1), "/", as.character(r.2)))]] <- diff.sym$vertex.diff
+    graphdata[[as.character(str_c(as.character(r.1), "/", as.character(r.2)))]] <- igraphs
+  }
+  compare.ranges$id <- 1:len
+  merged <- merge(compare.ranges, overview, by="id", all=T)
+  return (list(overview=merged, vertexdata=vertexdata, graphs=graphdata))
```

Comment:
This is the code I used for the data in the email, I figured it would be a nice addition.